guide.md

```diff
@@ -0,0 +1,350 @@
+# A guide to rules_nixpkgs
+
+TODO: does use flake work with vanilla direnv? nix-direnv is just an improvement over that, i believe
+TODO: use the same nixpkgs.nix for nixpkgs_git_repository and shell.nix
+TODO: is the flake shell pure? because if this is run on a non-nixos system, and /usr/bin/gcc continues to be visible, that messes things up
```

Comment:
If the Nix shell isn't pure, some of the stuff I've written about `@local_config_cc` not working when in the shell and having that extra insulation from accidentally using the non-hermetic toolchain becomes false, doesn't it?

Reply:
True, best to double check with `#nix`. I don't see `--pure` flag similar to the `nix-shell` one [on `nix develop`](https://nixos.org/manual/nix/unstable/command-ref/new-cli/nix3-develop.html), only `--impure`, but IIUC that is about Nix evaluation rather than the resulting shell environment.